contracts/pcv/maker/MakerRouter.sol

```diff
@@ -0,0 +1,229 @@
+import {Constants} from "../../Constants.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+/// @notice This contracts allows for swaps between FEI-DAI and FEI-USDC
+/// by using the FEI FEI-DAI PSM and the Maker DAI-USDC PSM
+/// @author k-xo
+contract MakerRouter is IMakerRouter, CoreRef {
+    using SafeERC20 for IERC20;
+
+    /// @notice reference to the Maker DAI-USDC PSM that this router interacts with
+    /// @dev points to Makers DssPsm contract
+    IDSSPSM public immutable daiPSM;
+
+    /// @notice reference to the FEI FEI-DAI PSM that this router interacts with
+    IPegStabilityModule public immutable feiPSM;
+
+    /// @notice reference to the DAI contract used.
+    /// @dev Router can be redeployed if DAI address changes
+    IERC20 public immutable dai;
+
+    /// @notice reference to the FEI contract used.
+    /// @dev Router can be redeployed if FEI address changes
+    IERC20 public immutable fei;
+
+    /// @notice scaling factor for USDC
+    uint256 public constant USDC_SCALING_FACTOR = 1e12;
+
+    constructor(
+        address _core,
+        IDSSPSM _daiPSM,
+        IPegStabilityModule _feiPSM,
+        IERC20 _dai,
+        IERC20 _fei
+    ) CoreRef(_core) {
+        daiPSM = _daiPSM;
+        feiPSM = _feiPSM;
+
+        dai = _dai;
+        fei = _fei;
+
+        fei.approve(address(feiPSM), type(uint256).max);
+        dai.approve(address(daiPSM), type(uint256).max);
+    }
+
+    /// @notice Function to swap from FEI to DAI
+    /// @param amountFeiIn the amount of FEI to be deposited
+    /// @param minDaiAmountOut the minimum amount of DAI expected to be received
+    /// @param to the address the DAI should be sent to once swapped
+    function swapFeiForDai(
+        uint256 amountFeiIn,
+        uint256 minDaiAmountOut,
+        address to
+    )
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        _redeemFromFeiPSM(amountFeiIn, minDaiAmountOut, to);
+    }
+
+    /// @notice Function to swap all of FEI balance to DAI
+    /// @param to the address the DAI should be sent to once swapped
+    function swapAllFeiForDai(address to)
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        _redeemAllBalanceFromFeiPSM(to);
+    }
+
+    /// @notice Function to swap from FEI to DAI
+    /// @dev Function will swap from FEI to DAI first then DAI to USDC
+    /// @param amountFeiIn the amount of FEI to be deposited
+    /// @param minDaiAmountOut the minimum amount of DAI expected to be received
+    /// @param to the address the DAI should be sent to once swapped
+    function swapFeiForUsdc(
+        uint256 amountFeiIn,
+        uint256 minDaiAmountOut,
+        address to
+    )
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        _redeemFromFeiPSM(amountFeiIn, minDaiAmountOut, address(this));
```

Comment:
As discussed, add a check to ensure `minDai` is >= USDC_SCALING_FACTOR, otherwise it will be incorrectly rounded down to 0

Reply:
check has been added in `_redeemFromFeiPSM`